### PR TITLE
Move test spec file

### PR DESCRIPTION
**Pull request by kirklandsign**
Use the new minibench for both LLM and generic model benchmarking.

### CODE REVIEW

**Comment by kirklandsign** [comment]
Later, could use @huydhn 's help with extracting generic model, and run it with org.pytorch.minibench/org.pytorch.minibench.BenchmarkActivity

**Review by huydhn — APPROVED** [review]

**Review by guangy10 — APPROVED** [review]

File: .github/workflows/upload-android-test-specs.yml [hunk 1]
```diff
@@ -4,13 +4,13 @@ on:
   pull_request:
     paths:
       - .github/workflows/upload-android-test-specs.yml
-      - examples/demo-apps/android/LlamaDemo/android-llm-device-farm-test-spec.yml
+      - extension/android/benchmark/android-llm-device-farm-test-spec.yml
```
Comment (guangy10) [inline .github/workflows/upload-android-test-specs.yml:7]: should we rename it as we are going to use it for both llm and non-genai mdoels

Comment (kirklandsign) [reply .github/workflows/upload-android-test-specs.yml:7]: Good point! Honestly there are more stuff to modify in this yml, so I will do it later together.

File: .github/workflows/android-perf.yml [hunk 2]
```diff
@@ -291,7 +291,7 @@ jobs:
       # Unlike models there are limited numbers of build flavor for apps, and the model controls whether it should build with bpe/tiktoken tokenizer.
       # It's okay to build all possible apps with all possible flavors in job "build-llm-demo". However, in this job, once a model is given, there is only
       # one app+flavor that could load and run the model.
-      android-app-archive: https://gha-artifacts.s3.amazonaws.com/${{ github.repository }}/${{ github.run_id }}/artifact/llm_demo/app-debug.apk
+      android-app-archive: https://gha-artifacts.s3.amazonaws.com/${{ github.repository }}/${{ github.run_id }}/artifact/minibench/app-debug.apk
       android-test-archive: https://gha-artifacts.s3.amazonaws.com/${{ github.repository }}/${{ github.run_id }}/artifact/llm_demo/app-debug-androidTest.apk
```
Comment (huydhn) [inline .github/workflows/android-perf.yml:295]: There is another one here :)

Comment (guangy10) [inline .github/workflows/android-perf.yml:295]: what about this line?

Comment (kirklandsign) [reply .github/workflows/android-perf.yml:295]: I didn't prepare for the instrumentation test for this app. So it's really a no-op if we use that. Do you think we should just no-op here?

Comment (kirklandsign) [reply .github/workflows/android-perf.yml:295]: I didn't prepare for the instrumentation test for this app. So it's really a no-op if we use that. Do you think we should just no-op here?

Comment (guangy10) [reply .github/workflows/android-perf.yml]: Let's put a note/comment to avoid confusion in the future

**Pull request closed without merging**